### ISSUE TITLE
Init a JSONArray with "[" will induce StackOverflowError in version 20171018.

### ISSUE THREAD

**Issue by yaosiang** [stleary/JSON-java#388]
When I upgrade org.json from 20170516 to 20171018, my unit test begin to failed, the test case shows if I init a JSONArray with "[", it will induce StackOverflowError.

```java
new JSONArray("[");
```

The call stack shows a infinite loop between JSONTokener.nextValue and JSONArray.\<init\>

```text
Exception in thread "main" java.lang.StackOverflowError
	at org.json.JSONTokener.back(JSONTokener.java:109)
	at org.json.JSONTokener.nextValue(JSONTokener.java:433)
	at org.json.JSONArray.<init>(JSONArray.java:118)
	at org.json.JSONTokener.nextValue(JSONTokener.java:434)
	at org.json.JSONArray.<init>(JSONArray.java:118)
	at org.json.JSONTokener.nextValue(JSONTokener.java:434)
	at org.json.JSONArray.<init>(JSONArray.java:118)
	at org.json.JSONTokener.nextValue(JSONTokener.java:434)
	at org.json.JSONArray.<init>(JSONArray.java:118)
	at org.json.JSONTokener.nextValue(JSONTokener.java:434)
	at org.json.JSONArray.<init>(JSONArray.java:118)
        ... skip
```

**Comment by johnjaylward** [comment]
This was fixed in #373

**Comment by yaosiang** [comment]
Got it. Hope this fix will release soon :)
